packages/lmi/src/lmi/llms.py

```diff
@@ -701,6 +704,44 @@ async def check_rate_limit(self, token_count: float, **kwargs) -> None:
                 **kwargs,
             )
 
+    async def _handle_refusal(
+        self, messages: list[Message], kwargs: dict[str, Any]
+    ) -> list[LLMResult]:
+        # Let's remove the current model from the configuration
+        current_model = self.name
+
+        new_model_list = [
+            model
+            for model in self.config["model_list"]
+            if current_model not in model.get("model_name")
+        ]
+
+        # Let's get any fallback we have defined anywhere and change it
+        new_fallbacks = []
+        for fallback in kwargs.get("fallbacks", []):
+            if current_model not in fallback:
+                new_fallbacks.append(fallback)
+            else:
+                target_model = fallback[current_model][0]
+                new_fallbacks.append({target_model: fallback[current_model][1:]})
+
+        # Now we update the configuration
+        # Here I am resetting all the configuration I think the user needs to set
+        # But it is possible there is a user misconfiguration that I'd be fixing here
+        self.name = new_model_list[0].get("model_name")
+        self.config["model_list"] = new_model_list
+        self.config["fallbacks"] = new_fallbacks
+        if "fallbacks" in kwargs:
+            kwargs["fallbacks"] = new_fallbacks
+        self.config["router_kwargs"] = self.config["router_kwargs"] | {
+            "set_verbose": True,
+            "fallbacks": new_fallbacks,
+        }
+        # Killing the router will force lmi to rebuild it
```

Comment:
```suggestion
        # Dereferencing the router will force lmi to rebuild it
```

Imo "killing" implies a teardown process, here we're just discarding it